runner/runner.go

```diff
@@ -231,59 +240,92 @@ func (r *Runner) run(ctx context.Context, checkMap map[string]*Check, list bool,
 		}
 	}
 
+	errs := make(chan responseError, len(currentCheckList))
+	responses := make(chan *Response, len(currentCheckList))
+
 	// main loop to get the checks info.
 	for _, name := range currentCheckList {
-		resp := &Response{
-			name: name,
-		}
 
-		currentCheck, ok := checkMap[name]
-		if !ok {
-			combinedResponse.errs["check not found"] = resp
-			combinedResponse.checkNotFound = true
-			continue
-		}
+		go func(name string) {
+			currentCheck, ok := checkMap[name]
+			if !ok {
+				errs <- responseError{"Check not found", name, true, nil}
+				return
+			}
 
-		// find runner for the given role only.
-		if !currentCheck.verifyRole(r.role) {
-			continue
-		}
+			// find runner for the given role only.
+			if !currentCheck.verifyRole(r.role) {
+				responses <- nil
+				return
+			}
 
-		var (
-			combinedOutput []byte
-			code           int
-			err            error
-			checkDuration  string
-		)
+			if _, ok := combinedResponse.checks[name]; ok {
```

Comment:
Maybe we should move this check to the for loop by deduplicating `currentCheckList` before we start looping?

Reply:
Checking `combinedResponse.checks` here is a data race: the on lines 303+ modify this map. I think we should just range through `currentCheckList` on line 246 (before entering the for loop) and remove duplicates.